src/types.ts

```diff
@@ -51,3 +51,9 @@ export type DefaultKey<Key extends string> = Record<'default', readonly [Key]>;
 
 export type QueryKeyFactoryResult<Key extends string, FactorySchema extends FactoryObject> = DefaultKey<Key> &
   FactoryOutput<Key, FactorySchema>;
+
+export type inferQueryKeys<FactorySchema> = {
+  [K in keyof FactorySchema]: FactorySchema[K] extends (...args: Array<any>) => ReadonlyArray<unknown>
```

Comment:
```suggestion
  [K in keyof FactorySchema]: FactorySchema[K] extends (...args: any[]) => readonly any[]
```
nit: I still have to update the linting config, but I wanna keep the `Array<>` syntax only for types that have unions like `Array<string | number>`

Reply:
Also, can you switch to using `P` (as in `Property`) instead of `K` to match the `FactoryOutput` implementation? Since `K` resembles `Key`, and `Key` is used throughout the code to identify the Key of a feature I wanna reduce the room for confusion